### PR TITLE
`init`: cover remote project cases

### LINKED ISSUE #126: `init`: cover remote project cases (in guide and ref)
There should probably be a separate section (in [Remote Objects](https://mlem.ai/doc/user-guide/remote-projects)) to explain that you can initialize MLEM projects in remote locations and give some motivation (why would I do that). The cmd ref. also should be much more emphatic about this.

_Originally posted by @jorgeorpinel in https://github.com/iterative/mlem.ai/pull/113#discussion_r915293079_

### PR DESCRIPTION
**Pull request by jorgeorpinel**
Closes #126

> There should probably be a separate section (in [Remote Objects](https://mlem.ai/doc/user-guide/remote-projects)) to explain that you can initialize MLEM projects in remote locations and give some motivation (why would I do that). The cmd ref. also should be much more emphatic about this.

### CODE REVIEW

**Comment by jorgeorpinel** [comment]
@aguschin @madhur-tandon can one of you take this over (after #113 is merged)? I'm happy to help edit any new text you add here.

**Review by aguschin — APPROVED** [review]

**Comment by aguschin** [comment]
@jorgeorpinel, looks like I cannot add you as a reviewer. Could you please review my changes manually?

File: content/docs/command-reference/init.md [hunk 1]
```diff
@@ -7,19 +7,29 @@ Initialize a MLEM project.
...
+The existence of a valid `.mlem/` directory in any location (including [remote])
+allows MLEM to store a references to MLEM objects stored in the project,
+enabling `$ mlem list` command, and configure MLEM
+[to work with DVC](http://localhost:8000/doc/use-cases/dvc) by setting this in
+`.mlem/config.yaml`.
+
+<admon type="tip">
+
+We recommend initializing MLEM projects inside Git repositories to track changes
+using standard Git workflows.
```
Comment (aguschin) [inline content/docs/command-reference/init.md:28]: What was the idea behind this? What Git workflows you assumed this enables? @jorgeorpinel

Comment (jorgeorpinel) [reply content/docs/command-reference/init.md:28]: > What was the idea behind this?

@aguschin I just moved this sentence from the previous paragraph into an admonition. Currently it reads

> Although we recommend using MLEM within a Git repository to track changes using
the standard Git workflows, ...

File: content/docs/command-reference/init.md [hunk 2]
```diff
@@ -7,19 +7,29 @@ Initialize a MLEM project.
...
 
-Although we recommend using MLEM within a Git repository to track changes using
-the standard Git workflows, this is not required. The existence of a `.mlem/`
-directory in any path (including remote) constitutes a MLEM project, and MLEM
-will be fully functional even without incorporating Git in one's workflow.
+The existence of a valid `.mlem/` directory in any location (including [remote])
+allows MLEM to store a references to MLEM objects stored in the project,
+enabling `$ mlem list` command, and configure MLEM
+[to work with DVC](http://localhost:8000/doc/use-cases/dvc) by setting this in
+`.mlem/config.yaml`.
```
Comment (jorgeorpinel) [inline content/docs/command-reference/init.md:23]:
```suggestion
The existence of a valid `.mlem/` directory in any location (including [remote])
enables all of MLEM's functions. Specifically, it allows for storing references
to MLEM objects found in the project (required by `mlem list`) as well as to
[integrate with DVC](/doc/use-cases/dvc).
```

File: content/docs/user-guide/remote-objects.md [hunk 3]
```diff
@@ -13,10 +13,21 @@ operations apply to any [object type] and location.
...
+## Initializing MLEM Project
+
+Although you can store MLEM objects in a Git repo or in a Cloud bucket without
+creating a MLEM project with `$ mlem init`, doing so enables you to organize and
+access your MLEM objects easier.
+
+This creates a `.mlem/` directory in the desired path. By default MLEM will save
+models to `.mlem/model/` and data to `.mlem/data`. If you save MLEM models
+outside of `.mlem/` folder, MLEM will store the reference to it inside `.mlem/`
+folder. This approach allows MLEM to list your MLEM objects in a project.
```
Comment (jorgeorpinel) [inline content/docs/user-guide/remote-objects.md:25]:
```suggestion
## Remote MLEM projects

Although you can store MLEM objects in any location such as a Git repo, Cloud
storage, or external drives, creating a MLEM project lets you
organize and [discover] MLEM objects consistently.

To create a MLEM project in a remote location, you can provide its URL or path
to `mlem init`.

[discover]: /doc/command-reference/list
```

**Review by jorgeorpinel — COMMENTED** [review]
> I cannot add you as a reviewer. Could you please review

I can't be a reviewer since I opened the PR 😋 so yes thanks for the reminder. Here's another iteration (committing):

File: content/docs/user-guide/remote-objects.md [hunk 4]
```diff
@@ -13,10 +13,19 @@ operations apply to any [object type] and location.
...
+
+To create a MLEM project in a remote location, you can provide its URL or path
+to `mlem init`.
+
 ## Listing objects
 
-You can list MLEM objects inside a remote MLEM project (e.g. in a Git
-repo) with `mlem list`. There's no need to download/clone the project.
+You can list MLEM objects inside a remote MLEM project (e.g. in a Git repo) with
+`mlem list`. There's no need to download/clone the project.
```
Comment (jorgeorpinel) [inline content/docs/user-guide/remote-objects.md:28]: Question @aguschin : There is some missing link between creating a remote project and having objects in there. What if there are objects in a remote location already, can you start a project on top of them and get them indexed automatically? I'm trying to figure out the motivation behind accepting URLs in `mlem init` and how that is relevant for this guide. Thanks

File: content/docs/command-reference/init.md (hunk 1 above)
Comment (aguschin) [reply content/docs/command-reference/init.md:27]: Ok, I think I got it now. I think it's clearer to say it like that. WDYT?
```suggestion
We recommend initializing MLEM projects inside Git repositories to track changes and
act on them using standard Git workflows.
```

File: content/docs/user-guide/remote-objects.md (hunk 4 above)
Comment (aguschin) [reply content/docs/user-guide/remote-objects.md:28]: > What if there are objects in a remote location already, can you start a project on top of them and get them indexed automatically?

No, currently they won't be indexed if you start a project in that location. But we should implement this I think.

Comment (aguschin) [reply content/docs/user-guide/remote-objects.md:28]: But we're discussing removing the indexing functionality from mlem completely, so let's write it as it is. Once we do that, we'll update the docs substantially anyway.

Comment (jorgeorpinel) [reply content/docs/user-guide/remote-objects.md:28]: OK thanks for that answer. It's still unclear to me why `mlem init` accepts URLs. If you have more context please share and/or update this page again. Merging the changes so far for now.

**Pull request merged**